packages/cache-handler/src/handlers/redis-stack.ts

```diff
@@ -98,42 +112,20 @@ export default async function createHandler<T extends RedisClientType>({
                 ? client.expireAt(options, keyPrefix + key, cacheHandlerValue.lifespan.expireAt)
                 : undefined;
 
-            await Promise.all([setCacheValue, expireCacheValue, setTags]);
+            await Promise.all([setCacheValue, expireCacheValue]);
         },
         async revalidateTag(tag) {
             assertClientIsReady();
 
-            const remoteTags: Record<string, string> = await client.hGetAll(
-                getTimeoutRedisCommandOptions(timeoutMs),
-                keyPrefix + sharedTagsKey,
-            );
-
-            const tagsMap = new Map(Object.entries(remoteTags));
-
-            const keysToDelete = [];
-
-            const tagsToDelete = [];
+            const query = await client.ft.search('idx:tags', `@tag:(${sanitizeTag(tag)})`);
 
-            for (const [key, tagsString] of tagsMap) {
-                const tags = JSON.parse(tagsString);
-
-                if (tags.includes(tag)) {
-                    keysToDelete.push(keyPrefix + key);
-                    tagsToDelete.push(key);
-                }
-            }
-
-            if (keysToDelete.length === 0) {
-                return;
-            }
+            const keysToDelete = query.documents.map((document) => document.id);
 
```

Comment:
Let's add check for `keysToDelete.length`. If there is no keys to revalidate there is no need to call `client.del`.

```js
if (keysToDelete.length === 0) {
    return;
}
```